stdlib/types/strings.nix

```
@@ -0,0 +1,72 @@
+{ ... }:
+rec {
+  isType = builtins.isString;
+  toType = builtins.toString;
+
+  sub = builtins.substring;
+
+  append = a: b: a + b;
+
+  # [string] -> string
+  concat = concatSep "";
+
+  # string -> [string] -> string
+  concatSep = builtins.concatStringsSep;
```

I was always wondering why it would not just be called this?

```suggestion
  join = builtins.concatStringsSep;
```